patches/minecraft/net/minecraft/world/entity/LivingEntity.java.patch

```diff
@@ -489,6 +489,17 @@
           if (this.f_20935_.m_41781_()) {
              this.m_21329_();
           }
+@@ -2951,6 +_,10 @@
+          this.m_21155_(1, false);
+       }
+ 
++      if (!this.f_20935_.m_41619_())
++      {
++         this.f_20935_.onStopUsingItem(this);
++      }
```

Comment:
You can keep this patch smaller by either putting this on one line or moving the if statement into the IForgeItemStack method.
Although I now realize that my point about this being called every time is moot because you call it on the currently used item, which will be EMPTY at that point and do nothing. So sorry, but you can revert adding this if statement

Reply:
I think saving two function calls every time a slot is switched is still worth it probably.

Reply:
Okay, fair enough